src/3d/StaticAssertions.fst

```diff
@@ -43,36 +55,62 @@ let compute_static_asserts (benv:B.global_env)
     match r with
     | None -> empty_static_asserts
     | Some r -> 
-      let sizeof_assertions =
+      let static_assertions =
         r.type_map
-        |> List.map
+        |> List.collect
           (fun (i, jopt) ->
             let j =
               match jopt with
               | None -> i
               | Some j -> j
             in
+            let offsets = TypeSizes.field_offsets_of_type env j in
+            let offset_of_assertions =
+              match offsets with
+              | Inr msg ->
+                Ast.warning 
+                  (Printf.sprintf
+                    "No offsetof assertions for type %s because %s\ns"
+                    (ident_to_string j)
+                    msg)
+                  i.range;
+                []
+              | Inl offsets ->
+                offsets |>
+                List.collect
+                  (fun offset -> 
+                    if TypeSizes.is_alignment_field (fst offset)

```

Comment:
Exclude assertions on fields that are added for alignment padding; they are implicit in the C type